### PR TITLE
Add support for custom errors in the `throw-new-error` rule

### LINKED ISSUE #4: Add option to the `throw-new-error` rule for working with custom errors
From: https://github.com/sindresorhus/eslint-plugin-xo/commit/53b7133c910695557d9c4a10079fac791b32c4dc#commitcomment-17077491

> An option that accepts an array of string/regex of error names? Could default to [/Error$/].
## 

New to making ESLint rules? Check out this quick guide on [how to get started](https://gist.github.com/sindresorhus/1656c46f23545deff8cc713649dcff26).

### PR DESCRIPTION
**Pull request by liubinyi**
added customer error by using /^(?:[A-Z][a-z0-9]*)*Error$/

fixes #4

### CODE REVIEW

File: rules/throw-new-error.js [hunk 1]
```diff
@@ -9,13 +9,14 @@ const errorTypes = [
 	'TypeError',
 	'URIError'
 ];
+const customerError = /^(?:[A-Z][a-z0-9]*)*Error$/;
```
Comment (jfmengels) [inline rules/throw-new-error.js]: It's actually a "custom" error, not a "customer" error.

File: rules/throw-new-error.js [hunk 2]
```diff
@@ -9,13 +9,14 @@ const errorTypes = [
...
 
 const create = context => ({
 	ThrowStatement: node => {
 		const arg = node.argument;
 		const error = arg.callee;
 
-		if (arg.type === 'CallExpression' && errorTypes.indexOf(error.name) !== -1) {
+		if (arg.type === 'CallExpression' && (errorTypes.indexOf(error.name) !== -1 || customerError.test(error.name))) {
```
Comment (jfmengels) [inline rules/throw-new-error.js]: If the regex is well made, we don't need the `errorTypes.indexOf(error.name) !== -1` check anymore. If you remove it and all tests pass, it should be fine to remove.

File: test/throw-new-error.js [hunk 3]
```diff
@@ -14,7 +14,8 @@ ruleTester.run('new-error', rule, {
 	valid: [
 		'throw new Error()',
 		'new Error()',
-		'throw new TypeError()'
+		'throw new TypeError()',
+		'throw new CustomerError()'
```
Comment (jfmengels) [inline test/throw-new-error.js]: everywhere you use "customer", replace by "custom"

File: rules/throw-new-error.js (hunk 1 above)
Comment (jfmengels) [inline rules/throw-new-error.js]: Is there a use for the `?:`? I'm not sure what it does.

File: test/throw-new-error.js [hunk 4]
```diff
@@ -28,6 +29,11 @@ ruleTester.run('new-error', rule, {
 			errors
 		},
 		{
+			code: `throw CustomerError('foo')`,
+			output: `throw new CustomerError('foo')`,
+			errors
+		},
```
Comment (jfmengels) [inline test/throw-new-error.js]: Can you add a few tests that make sure the regex is well built? I'm thinking things like the following, but add more if you think of any

Valid
```
throw fooBarBazError('foo') // does not start with an upper case character
```

Invalid
```
throw FooBarBazError('foo') // With multiple upper case characters
throw ABCError('foo') // With multiple upper case characters in a row
throw Abc3Error('foo') // With numbers
```

File: rules/throw-new-error.js (hunk 1 above)
Comment (liubinyi) [inline rules/throw-new-error.js]: will update them:
i believe ?: is look ahead expression.

File: test/throw-new-error.js (hunk 4 above)
Comment (liubinyi) [inline test/throw-new-error.js]: sure.

File: rules/throw-new-error.js (hunk 2 above)
Comment (liubinyi) [inline rules/throw-new-error.js]: ah yeah, i will update it.

**Comment by liubinyi** [comment]
@jfmengels Thanks for reviewing and providing useful feedback! I updated the PR base on you comments. i will update it again once other questions are cleared out.

File: rules/throw-new-error.js [hunk 5]
```diff
@@ -1,21 +1,13 @@
...
-const errorTypes = [
-	'Error',
-	'EvalError',
-	'RangeError',
-	'ReferenceError',
-	'SyntaxError',
-	'TypeError',
-	'URIError'
-];
+const customeError = /^(?:[A-Z][a-z0-9]*)*Error$/;
```
Comment (sindresorhus) [inline rules/throw-new-error.js]: `customeError` => `customError`

Comment (liubinyi) [inline rules/throw-new-error.js]: update coming soon!

Comment (liubinyi) [inline rules/throw-new-error.js]: @sindresorhus thanks for reviewing, updated.

**Comment by sindresorhus** [comment]
Looks good. Thank you! :)

File: rules/throw-new-error.js (hunk 1 above)
Comment (kj35878) [reply rules/throw-new-error.js]: > 将更新它们。 I believe ？： 表示该组已匹配，但未被捕获以进行反向引用
我是俊爷，你300多RMB礼物再来一份，不准再坑了，联系方式，手机+8619372278999，我有很多问题问你，看在6年朋友的份上，马上联系我，我得推特是kuangjun，头像是一条龙。脸书联系不上你。

**Pull request merged**